airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2773,6 +2773,14 @@ def _submit_job(
         hook: BigQueryHook,
         job_id: str,
     ) -> BigQueryJob:
+        # Annotate the job with dag and task id labels
+        if "labels" in self.configuration:
+            if isinstance(self.configuration["labels"], dict):
+                self.configuration["labels"]["airflow-dag"] = self.dag_id
+                self.configuration["labels"]["airflow-task"] = self.task_id
```

Comment:
Do labels support all chars?

Reply:
https://cloud.google.com/bigquery/docs/labels-intro#requirements
Good point, there are some restrictions outline here.  I'll take a look.

Reply:
I've added length and content checks according to these docs.  The labels won't be added unless everything is valid.